cli/studio.py

```diff
@@ -64,9 +75,27 @@ def _build_single_scenario(clean: bool, allow_offset_map: bool, scenario: str):
     scenario_root_str = str(scenario_root)
 
     scenario_py = scenario_root / "scenario.py"
+
     if scenario_py.exists():
-        _install_requirements(scenario_root)
-        subprocess.check_call([sys.executable, "scenario.py"], cwd=scenario_root)
+        copy_file = None
+        try:
+            if seed is not None:
+
+                copy_file = smarts.core.gen_id() + ".py"
+                scenario_py_copy = scenario_root / copy_file
+                with open(scenario_py_copy, "w") as c:
+                    with open(scenario_py, "r") as o:
+                        c.write(f"import smarts.core; smarts.core.seed({seed});\n")
```

Comment:
This makes me a bit uncomfortable.

Why can't we just use the existing `seed` argument that `gen_scenario()` already takes and add a call there (within `gen_scenario()`) to `smarts.core.seed(seed)`?

(Alternatively, we could just _document_ that if scenario authors want consistency, they can call `smarts.core.seed()` themselves, and even show some examples, like showing what we have in `scenarios/sumo/loop/scenario.py`.)

Reply:
I understand the discomfort, the intention here is that the default behaviour should be predictable and that random behaviour should be opt-in. It is still possible for a user to get random values if they `smarts.core.seed(seed)` in `scenario.py` because that would occur afterwards.